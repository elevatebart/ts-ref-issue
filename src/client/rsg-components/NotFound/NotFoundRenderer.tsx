import React from 'react';
import PropTypes from 'prop-types';
import Markdown from 'rsg-components/Markdown';
import Styled, { JssInjectedProps } from 'rsg-components/Styled';
import * as Rsg from '../../../typings';

const styles = ({ maxWidth }: Rsg.Theme) => ({
	root: {
		maxWidth,
		margin: [[0, 'auto']],
	},
});

export const NotFoundRenderer: React.FunctionComponent<JssInjectedProps> = ({ classes }) => {
	return (
		<div className={classes.root}>
			<Markdown
				text={`
# Page not found
The link you followed may be broken, or the page may have been removed.
`}
			/>
		</div>
	);
};

NotFoundRenderer.propTypes = {
	classes: PropTypes.objectOf(PropTypes.string.isRequired).isRequired,
};

export default Styled(styles)(NotFoundRenderer);
