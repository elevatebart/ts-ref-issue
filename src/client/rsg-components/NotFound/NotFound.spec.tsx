import React from 'react';
import { shallow } from 'enzyme';
import { NotFoundRenderer } from './NotFoundRenderer';

it('renderer should render not found message', () => {
	const actual = shallow(<NotFoundRenderer classes={{}} />);

	expect(actual).toMatchSnapshot();
});
