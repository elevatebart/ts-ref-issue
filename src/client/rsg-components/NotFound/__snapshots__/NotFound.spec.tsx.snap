// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderer should render not found message 1`] = `
<div>
  <Markdown
    text="
# Page not found
The link you followed may be broken, or the page may have been removed.
"
  />
</div>
`;
